envydis/gm107.c

```diff
@@ -36,14 +36,17 @@
 	{ 0, 0, OOPS } \
 }
 
-/* sched fields */
-static struct bitfield sched0_bf = {  0, 21 };
-static struct bitfield sched1_bf = { 21, 21 };
-static struct bitfield sched2_bf = { 42, 21 };
-
-#define SCHED0 atomrimm, &sched0_bf
-#define SCHED1 atomrimm, &sched1_bf
-#define SCHED2 atomrimm, &sched2_bf
+/* sched fields (source:  https://github.com/NervanaSystems/maxas/wiki/Control-Codes) */
+#define SCHED_STALL(n)     atomrimm, (struct bitfield[]) { { n * 21 + 0,  4 } } /* stall counts */
+#define SCHED_YIELD(n)     atomrimm, (struct bitfield[]) { { n * 21 + 4,  1 } } /* yield hint flag */
+#define SCHED_BAR_WRITE(n) atomrimm, (struct bitfield[]) { { n * 21 + 5,  3 } } /* write dep bar */
+#define SCHED_BAR_READ(n)  atomrimm, (struct bitfield[]) { { n * 21 + 8,  3 } } /* read dep bar */
+#define SCHED_BAR_WAIT(n)  atomrimm, (struct bitfield[]) { { n * 21 + 11, 6 } } /* wait dep bar */
+#define SCHED_REUSE(n)     atomrimm, (struct bitfield[]) { { n * 21 + 17, 4 } } /* reuse flag */
+#define SCHED(n)           static struct insn tabsched##n[] = { 0, 0, N("{"), SCHED_YIELD(n), SCHED_STALL(n), SCHED_BAR_WRITE(n), SCHED_BAR_READ(n), SCHED_BAR_WAIT(n), SCHED_REUSE(n),N("}") }
```

Comment:
As I mentioned in the other change ... please try to use SESTART/SEEND, which is how we handle groups everywhere else.

I also wonder if it'd be useful to add semantic names to each flag, like

stl 0x0 yld wr 0x0 rd 0x0 wt 0x0 reuse 0x0

or ... something like that. It's a lot longer, but a lot nicer than 6 random values.